controllers/common/common.go

```diff
@@ -245,6 +247,68 @@ func FullNameForPolicy(plc *policiesv1.Policy) string {
 	return plc.GetNamespace() + "." + plc.GetName()
 }
 
+// GetRepPoliciesInPlacementBinding returns a list of the replicated policies that are either direct subjects of
+// the given PlacementBinding, or are in PolicySets that are subjects of the PlacementBinding.
+// The list items are guaranteed to be unique (for example if a policy is in multiple sets).
+func GetRepPoliciesInPlacementBinding(
+	ctx context.Context, c client.Client, pb *policiesv1.PlacementBinding,
+) []reconcile.Request {
+	decisions, err := GetDecisions(ctx, c, pb)
+	if err != nil {
+		return []reconcile.Request{}
+	}
+	// Use this for removing duplicated policies
+	table := map[reconcile.Request]bool{}
+
+	for _, subject := range pb.Subjects {
+		if subject.APIGroup != policiesv1.SchemeGroupVersion.Group {
+			continue
+		}
+
+		switch subject.Kind {
+		case policiesv1.Kind:
+			for _, pd := range decisions {
+				req := reconcile.Request{NamespacedName: types.NamespacedName{
+					Name:      pb.GetNamespace() + "." + subject.Name,
+					Namespace: pd.ClusterName,
+				}}
+
+				table[req] = true
+			}
+
+		case policiesv1.PolicySetKind:
+			setNN := types.NamespacedName{
+				Name:      subject.Name,
+				Namespace: pb.GetNamespace(),
+			}
+
+			policySet := policiesv1beta1.PolicySet{}
+			if err := c.Get(ctx, setNN, &policySet); err != nil {
+				continue
+			}
+
+			for _, plc := range policySet.Spec.Policies {
+				for _, pd := range decisions {
+					req := reconcile.Request{NamespacedName: types.NamespacedName{
+						Name:      pb.GetNamespace() + "." + string(plc),
+						Namespace: pd.ClusterName,
+					}}
+
+					table[req] = true
+				}
+			}
+		}
+	}
+
+	result := []reconcile.Request{}
```

Comment:
This will save on memory allocations and copies.
```suggestion
	result := make([]reconcile.Request, 0, len(table))
```